Audio/BnkCompiler/ObjectConfiguration/Warhammer3/DialogueEventData.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Shared.Core.Misc;
+using Audio.Storage;
+
+namespace Audio.BnkCompiler.ObjectConfiguration.Warhammer3
+{
+    public class DialogueEventData
+    {
+        private readonly IAudioRepository _audioRepository;
+
+        public DialogueEventData(IAudioRepository audioRepository) => _audioRepository = audioRepository;
+
+        public static Dictionary<string, List<string>> ExtractedDialogueEvents = new Dictionary<string, List<string>>();
+        public static List<string> ExtractedStateGroups = new List<string>();
+        public static Dictionary<string, List<string>> ExtractedStates = new Dictionary<string, List<string>>();        
+
+        // Dynamically extract dialogue events from the master dat file (dialogue events are contained within Section 4).
+        public void ExtractDialogueEventsDataFromDat()
+        {
+            ExtractedDialogueEvents = new Dictionary<string, List<string>>();
+            ExtractedStateGroups = new List<string>();
+
+            var datDumpMasterPath = $"{DirectoryHelper.Temp}\\DatDumps\\dat_dump_master.txt";
+            var lines = File.ReadAllLines(datDumpMasterPath);
+            var datSection4 = ExtractSection(lines, "Section 4", "Section 5");
+
+            foreach (var line in datSection4)
+            {
+                var parts = ParseLine(line);
+                var extractedDialogueEvent = parts.Item1;
+                var extractedStateGroups = parts.Item2;
+
+                foreach (var extractedStateGroup in extractedStateGroups)
+                {
+                    if (extractedStateGroup != "")
+                    {
+                        var stateGroup = _audioRepository.GetNameFromHash(uint.Parse(extractedStateGroup));
+
+                        if (!ExtractedDialogueEvents.ContainsKey(extractedDialogueEvent))
+                            ExtractedDialogueEvents[extractedDialogueEvent] = new List<string>();
+
+                        ExtractedDialogueEvents[extractedDialogueEvent].Add(stateGroup);
+
+                        if (!ExtractedStateGroups.Contains(stateGroup))
+                            ExtractedStateGroups.Add(stateGroup);
+                    }
+                }
+            }
+
+            ExtractedStateGroups.Sort();
+
+            var csvFilePath = $"{DirectoryHelper.Temp}\\DatDumps\\dat_dump_dialogue_events.csv";
+            ExportData(ExtractedDialogueEvents, csvFilePath);
+        }
+
+        // Dynamically extract states from the master dat file (states are contained within Section 3).
+        public static void ExtractStatesDataFromDat()
+        {
+            ExtractedStates = new Dictionary<string, List<string>>();
+
+            var datDumpMasterPath = $"{DirectoryHelper.Temp}\\DatDumps\\dat_dump_master.txt";
```

Comment:
you can just get this directly from the dat file, no reason to go through a temp file? It will only be a source of error

Reply:
they are also found in the audio repository i think? They might need some labling for your use case though